test/integration/scheduler/predicates_test.go

```diff
@@ -1219,25 +1235,250 @@ func TestPodTopologySpreadFilter(t *testing.T) {
 			candidateNodes:   []string{"node-0", "node-1", "node-2", "node-3"},
 			enableMinDomains: true,
 		},
+		{
+			name: "NodeAffinityPolicy honored with labelSelectors, pods spread across zone as 2/1",
+			incomingPod: st.MakePod().Name("p").Label("foo", "").Container(pause).
+				NodeSelector(map[string]string{"foo": ""}).
+				SpreadConstraint(1, "zone", v1.DoNotSchedule, st.MakeLabelSelector().Exists("foo").Obj(), nil, nil, nil).
+				Obj(),
+			existingPods: []*v1.Pod{
+				st.MakePod().Name("p1a").Node("node-1").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Name("p2a").Node("node-2").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Name("p3a").Node("node-3").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Name("p4a").Node("node-4").Label("foo", "").Container(pause).Obj(),
+			},
+			fits: true,
+			specificNodes: []*v1.Node{
+				st.MakeNode().Name("node-1").Label("node", "node-1").Label("zone", "zone-1").Label("foo", "").Obj(),
+				st.MakeNode().Name("node-2").Label("node", "node-2").Label("zone", "zone-1").Label("foo", "").Obj(),
+				st.MakeNode().Name("node-3").Label("node", "node-3").Label("zone", "zone-2").Obj(),
+				st.MakeNode().Name("node-4").Label("node", "node-4").Label("zone", "zone-2").Label("foo", "").Obj(),
+			},
+			candidateNodes:             []string{"node-4"}, // node-3 is filtered out by NodeAffinity plugin
+			enableNodeInclustionPolicy: true,
+		},
+		{
+			name: "NodeAffinityPolicy ignored with nodeAffinity, pods spread across zone as 1/~2~",
```

Comment:
No need to separate nodeAffinity from node selectors. We run the same code for both.

Integration tests are expensive, so we should try to keep the tests minimal.

Reply:
One test enabled `NodeAffinityPolicy`, the other one disabled.